Forta-Bots-Challenge-3/src/utils.ts

```diff
@@ -0,0 +1,96 @@
+import { Finding, getAlerts, sendAlerts } from "forta-agent";
+import { providers, Contract, BigNumber } from "ethers";
+import { createEscrowBalanceFinding, createFinding } from "./finding";
+import { agentId } from "../forta.config.json";
+import { ERC20_ABI } from "./constants";
+
+const chainNames = {
+  1: "Ethereum",
+  10: "Optimism",
+  42161: "Arbitrum",
+};
+
+export type contractAddresses = {
+  l1DaiAddress: string;
+  l2DaiAddress: string;
+  optimismEscrow: string;
+  arbitrumEscrow: string;
+};
+
+export const emitEscrowBalance = async (
+  provider: providers.Provider,
+  blockNumber: number,
+  l1DaiAddress: string,
+  optimismEscrow: string,
+  arbitrumEscrow: string
+) => {
+  const l1DaiContract = new Contract(l1DaiAddress, ERC20_ABI, provider);
+
+  // call L1 DAI contract to get balances of each L2's escrow contract
+  const optimismEscrowBalance = await l1DaiContract.balanceOf(optimismEscrow, { blockTag: blockNumber });
+  const arbitrumEscrowBalance = await l1DaiContract.balanceOf(arbitrumEscrow, { blockTag: blockNumber });
+
+  // create finding
+  const escrowBalanceFinding = createEscrowBalanceFinding(
+    blockNumber.toString(),
+    optimismEscrowBalance,
+    arbitrumEscrowBalance
+  );
+
+  // send alert containing previously declared finding
+  sendAlerts({
+    botId: agentId,
+    finding: escrowBalanceFinding,
+  });
```

Comment:
Not sure why you chose `sendAlerts` over the usual `findings.push(..)`. I'm not very familiar with it, but it's a new SDK method meant to be used with those "new/V2/external" bots. (I tried running it and makes the bot crash for me)